scalafix-core/src/main/scala/scalafix/rewrite/NoAutoTupling.scala

```diff
@@ -15,32 +15,68 @@ case class NoAutoTupling(mirror: Mirror) extends SemanticRewrite(mirror) {
     ctx.addLeft(args.head.tokens.head, "(") +
       ctx.addRight(args.last.tokens.last, ")")
 
+  private[this] def fixMethods(ctx: RewriteCtx,
+                               jvmSignature: String,
+                               argss: Seq[Seq[Term.Arg]],
+                               fixed: mutable.Set[Seq[Term.Arg]]): Patch = {
+    val argListSignatures =
+      jvmSignature.stripPrefix("(").takeWhile(_ != ')').split(';')
+    argListSignatures
+      .zip(argss)
+      .foldLeft(Patch.empty) {
+        case (patches, (argListSignature, args)) =>
+          if (!fixed(args) && singleTuplePattern
+                .matcher(argListSignature)
+                .matches && args.length > 1) {
+            fixed += args // dirty hack, see explanation above
+            patches + addWrappingParens(ctx, args)
+          } else {
+            patches
+          }
+      }
+  }
+
+  private[this] def fixFunctions(ctx: RewriteCtx,
+                                 symbol: Symbol,
+                                 argss: Seq[Seq[Term.Arg]],
+                                 fixed: mutable.Set[Seq[Term.Arg]],
+                                 mirror: Mirror): Patch =
+    mirror.database.denotations
```

Comment:
FYI, this list currently only contains denotations from this compilation run. It won't have denotations for 3rd party libraries, but that is planned to change in the future.